google/cloud/security/common/gcp_type/firewall_rule.py

```diff
@@ -66,3 +105,359 @@ def priority(self):
+            action_dict = {
+                'firewall_rule_allowed': self.allowed,
+                'firewall_rule_denied': self.denied,
+            }
+            self._firewall_action = FirewallAction(**action_dict)
+        return self._firewall_action
+
+    def __lt__(self, other):
+        """Test whether this policy is contained in another policy.
+
+        Checks if this rule is a subset of the allowed/denied ports and
+        protocols that are in the other rule.
+
+        Args:
+          other(FirewallRule): object to compare to
+
+        Returns:
+          bool: comparison result
+        """
+        return (self.direction == other.direction and
+                self.network == other.network and
+                set(self.source_tags).issubset(other.source_tags) and
+                set(self.target_tags).issubset(other.target_tags) and
+                self.firewall_action < other.firewall_action and
+                ips_in_list(self.source_ranges, other.source_ranges) and
+                ips_in_list(self.destination_ranges, other.destination_ranges))
+
+    def __gt__(self, other):
+        """Test whether this policy contains the other policy.
+
+        Checks if this rule is a superset of the allowed/denied ports and
+        protocols that are in the other rule.
+
+        Args:
+          other(FirewallRule): object to compare to
+
+        Returns:
+          bool: comparison result
+        """
+        return (self.direction == other.direction and
+                self.network == other.network and
+                set(other.source_tags).issubset(self.source_tags) and
+                set(other.target_tags).issubset(self.target_tags) and
+                self.firewall_action > other.firewall_action and
+                ips_in_list(other.source_ranges, self.source_ranges) and
+                ips_in_list(other.destination_ranges, self.destination_ranges))
+
+    def __eq__(self, other):
+        """Test whether this policy is the same as the other policy.
+
+        Args:
+          other(FirewallRule): object to compare to
+
+        Returns:
+          bool: comparison result
+        """
+        return (self.direction == other.direction and
+                self.network == other.network and
+                self.source_tags == other.source_tags and
+                self.target_tags == other.target_tags and
+                self.source_ranges == other.source_ranges and
+                self.destination_ranges == other.destination_ranges and
+                self.firewall_action == other.firewall_action)
+
+    def is_equivalent(self, other):
+        """Test whether this policy is equivalent to the other policy.
+
+        Args:
+          other(FirewallRule): object to compare to
+
+        Returns:
+          bool: comparison result
+        """
+        return (self.direction == other.direction and
+                self.network == other.network and
+                self.source_tags == other.source_tags and
```

Comment:
Nit: This could use self._source_tags == other._source_tags and then it's an O(1) operation to compare two frozensets, which is just a hash value compare. Otherwise you are creating lists and comparing them.